### PR TITLE
DHFPROD-4659: Visualize metadata of the document in card view

### PR DESCRIPTION
**Pull request by timur-isangulov**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by Sanjeevani19 — CHANGES_REQUESTED** [review]
@timur-isangulov Could you please add unit tests for Docs with no metadata, just like you have added for docs with all metadata

**Comment by timur-isangulov** [comment]
> @timur-isangulov Could you please add unit tests for Docs with no metadata, just like you have added for docs with all metadata

Added, please review again.

**Comment by Sanjeevani19** [comment]
> > @timur-isangulov Could you please add unit tests for Docs with no metadata, just like you have added for docs with all metadata
> 
> Added, please review again.

@timur-isangulov  will there be a scenario where there are docs with partial metadata ? In that case we might need a test for that scenario too. Also the info tooltip doesn't stick to the card where it has been opened, as the page scrolls position also changes.

**Review by Sanjeevani19 — CHANGES_REQUESTED** [review]
@timur-isangulov will there be a scenario where there are docs with partial metadata ? In that case we might need a test for that scenario too. 
- Also the info tooltip doesn't stick to the card where it has been opened, as the page scrolls position also changes.

File: marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx [hunk 1]
```diff
@@ -67,6 +65,44 @@ const RawDataCardView = (props) => {
     }
   }
 
+  const displayDocumentSources = (item) => {
+    let sources = item.hubMetadata?.sources.map((el) => {
+      return el.name;
+    }).join(", ");
```
Comment (rjrudin) [inline marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx:71]: @timur-isangulov  @rahulvudutala  @yasht2010  Should consider that a source name could have a comma in it. We don't expect that, but it's also not prohibited. That's a reason why I favor some sort of structured grouping any time we list sources, as opposed to relying on textual delimiters.

File: marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx [hunk 2]
```diff
@@ -67,6 +65,44 @@ const RawDataCardView = (props) => {
...
+    return sources;
+  }
+
+  const emptyField = (
+    <p className={styles.textDisabled}>none</p>
+  )
+
+  const displayDocumentMetadata = (item) => {
```
Comment (rjrudin) [inline marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx:79]: Can you change this to "RecordMetadata"? We are trying to use the term "Record" instead of document, as Record is more familiar to a user. Additionally, "document metadata" is a specific thing in MarkLogic - https://docs.marklogic.com/xdmp.documentGetMetadata - and that differs from what we are defining as "metadata" here.

File: marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx [hunk 3]
```diff
@@ -67,6 +65,44 @@ const RawDataCardView = (props) => {
...
+      <div className={styles.popover} data-testid={item.uri + '-popover'}>
+        <div className={styles.colKey}>
+          <p>Source:</p>
+          <p>Flow:</p>
+          <p>Step:</p>
+          <p>Created On:</p>
+        </div>
+        <div className={styles.colValue}>
+          {item.hubMetadata?.sources ? <span className={styles.valText} data-testid={item.uri + '-sources'}>
+            <OverflowTooltip title={displayDocumentSources(item)} placement={"bottom"} content={displayDocumentSources(item).substring(0, 28)} width={"200px"} />
```
Comment (rjrudin) [inline marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx:90]: Likewise, "RecordSources" would be more appropriate here. 

In fact, I'd rather this component be called "record-view" instead of "raw-data-card". This data is not necessarily "raw", which we don't have a clear definition of either. It may be highly structured data, just not an entity instance.

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx (hunk 1 above)
Comment (rahulvudutala) [reply marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx:71]: @rjrudin This is just an overview of source names limited to 23 characters. The accurate Record metadata can be seen in the Detail view. I assume it should be alright for one edge case. 
@yasht2010 can confirm.

**Comment by timur-isangulov** [comment]
> @timur-isangulov will there be a scenario where there are docs with partial metadata ? In that case we might need a test for that scenario too.
> 
> * Also the info tooltip doesn't stick to the card where it has been opened, as the page scrolls position also changes.

The will be cases with particular metadata but since the two other cases are covered with existing and missing metadata probably it doesn't add much value. 
Fixed the issue with the popover. Thank you for noticing!

File: marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx (hunk 3 above)
Comment (timur-isangulov) [reply marklogic-data-hub-central/ui/src/components/raw-data/raw-data-card-view/raw-data-card-view.tsx]: Changed to RecordSources and renamed component. Thanks.

**Review by Sanjeevani19 — APPROVED** [review]

**Review by rahulvudutala — DISMISSED** [review]

**Review by xnikhil08 — CHANGES_REQUESTED** [review]
Just few CSS changes as mentioned below.
1. Change the cursor to pointer when user hovers over the metadata icon. It should be consistent with other clickable icons in the application.
2. Reduce the padding on the right(like the one in design ) so the text looks evenly spread in the popover.
3. Add just one more space between 'CreatedOn:' and the Date part after it.

**Comment by timur-isangulov** [comment]
Due to high number of file conflicts for several commits opened https://github.com/marklogic/marklogic-data-hub/pull/4716 with addressing Nikhil's comment.
